### PR TITLE
My Bcup

### PR DESCRIPTION
**Pull request by perry-mitchell**
Add support for My Buttercup!

Also removes explicit items for ownCloud/Nextcloud.

### CODE REVIEW

File: source/background/library/browserEvents.js [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
+    setAccessToken as setMyButtercupAccessToken,
+    setRefreshToken as setMyButtercupRefreshToken
+} from "../../shared/actions/myButtercup.js";
 
 const AUTOLOGIN_EXPIRY = ms("45s");
 const BUTTERCUP_DOMAIN_REXP = /^https:\/\/buttercup.pw\//;
 const DROPBOX_ACCESS_TOKEN_REXP = /access_token=([^&]+)/;
 const GOOGLE_DRIVE_AUTH_CODE_REXP = /\?googleauth&code=([^&#?]+)/;
+const MYBUTTERCUP_AUTH_CODE_REXP = /code=([^&]+)/;
+const MYBUTTERCUP_CALLBACK_URL_REXP = /^https?:\/\/(localhost:8000|my\.buttercup\.pw)\/oauth\/authorized\//;
```
Comment (sallar) [inline source/background/library/browserEvents.js]: Is it safe to leave this `localhost` thing here?

File: source/setup/containers/RemoteExplorer.js [hunk 2]
```diff
@@ -23,6 +24,8 @@ function contentsToTree(contents, fetchType) {
             return googleDriveContentsToTree(contents);
         case "localfile":
             return localContentsToTree(contents);
+        // case "mybuttercup":
```
Comment (sallar) [inline source/setup/containers/RemoteExplorer.js]: Extra comments here

File: source/shared/library/myButtercup.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export const MYBUTTERCUP_CLIENT_ID = "bcup_browser_ext";
+export const MYBUTTERCUP_CLIENT_SECRET = "71cf1f22921b5ceb84d7ea71d95a854b";
+// export const MYBUTTERCUP_REDIRECT_URI = "https://my.buttercup.pw/oauth/authorized/";
```
Comment (sallar) [inline source/shared/library/myButtercup.js]: And here

File: source/background/library/browserEvents.js (hunk 1 above)
Comment (perry-mitchell) [reply source/background/library/browserEvents.js]: Yep! It's just for development. Validation of the redirect is performed by the server anyway.

**Pull request merged**